packages/worker/src/workflow/vm-shared.ts

```diff
@@ -28,12 +29,15 @@ export function setUnhandledRejectionHandler(getWorkflowByRunId: (runId: string)
         return;
       }
     }
-    // The user's logger is not accessible in this thread,
-    // dump the error information to stderr and abort.
-    console.error('Unhandled rejection', { runId }, err);
-    process.exit(1);
+
+    console.error('An Unhandled Promise rejection could not be associated to a Workflow Run', { runId, error: err });
+    throw new UnhandledRejectionError(
```

Comment:
This error message is a bit cryptic, it could help say why this happens and how to fix it, or point to docs...

Reply:
That's a totally unexpected case. If a user ever see this, that's very likely a bug in the SDK. There's nothing more to document.